### ISSUE TITLE
Specify the path to the docker config file

### ISSUE THREAD

**Issue by msoub** [GoogleContainerTools/jib#1618]
I'd like to specify the path to my Docker config file, by default it reads it from `${user.home}/.docker/config.json`.

I can see from `CredentialRetrieverFactory#dockerConfig(java.nio.file.Path)` that it's possible, however the documentation does not specify how to do it.

Thank you for your help.

**Comment by dgageot** [comment]
Hi @msoub! Have you tried running mvn with `-Duser.home=[you decide]`?

Something like `mvn -Duser.home=/home jib:build`

**Comment by chanseokoh** [comment]
I believe `-Duser.home=/home` will make Maven cache artifacts under `/home/.m2`. (Correct me if I am wrong.) That is, it changes the user home JVM-wise.

**Comment by jontro** [comment]
I successfully worked around the issue.

Using jenkins docker.withRegistry will store the path in an environment variable. https://jenkins.io/doc/book/pipeline/docker/

Using kotlin dsl syntax:
```
jib {
    var credential: JavaOptional<com.google.cloud.tools.jib.api.Credential> = JavaOptional.empty()
    val path = System.getenv("DOCKER_CONFIG")
    if (path != null) {
        credential = com.google.cloud.tools.jib.registry.credentials.DockerConfigCredentialRetriever("<repository>", Paths.get(path + "/config.json")).retrieve {
            System.out.println(it.message)
        }
    }

    to {
        image = "<image>"
        if (credential.isPresent) {
            auth {
                username = credential.get().username
                password = credential.get().password
            }
        }
    }

}

```

**Comment by chanseokoh** [comment]
@jontro for Jenkins, another way is to use `withCredentials()` to pass credentials through environment variables. (See https://stackoverflow.com/a/59688329/1701388 for more details.)

```
    def dockerImage
    stage('publish docker') {
        withCredentials([usernamePassword(credentialsId: 'myregistry-login', passwordVariable: 'DOCKER_REGISTRY_PWD', usernameVariable: 'DOCKER_REGISTRY_USER')]) {
            // assumes Jib is configured to use the environment variables
            sh "./mvnw -ntp jib:build"
        }
    }
```

**Comment by jontro** [comment]
@chanseokoh Not sure how to use that with ecr since credentials are fetched using the aws api. This plugin https://wiki.jenkins.io/display/JENKINS/Amazon+ECR makes it easy to use with withRegistry

**Comment by jontro** [comment]
When thinking a bit more about it the environment variable DOCKER_CONFIG is the standard way of changing the location. The plugin could take this into account automatically and fall back to $HOME/.docker

**Comment by chanseokoh** [comment]
It'd be useful to fix the original issue (feature request to specify a Docker config path on the command line). For example, Tekton currently creates a file at `/tekton/home/.docker/config.json`, where `/tekton/home` is often not the home directory of a running user. (And for future versions, Tekton proposed a plan to use a [fixed directory](https://github.com/tektoncd/pipeline/issues/2013#issuecomment-596597742) instead of `/tekton/home` unrelated to any home.)

**Comment by TadCordle** [comment]
For now, I think it makes sense to just use the `DOCKER_CONFIG` environment variable at the plugin level (but keep things as they are in jib-core). In the future we can add a config parameter or system property if it makes sense.

**Comment by chanseokoh** [comment]
The environment variable makes sense.

> In the future we can add a config parameter or system property if it makes sense.

When deciding this, one factor is that this config is highly platform-dependent.

**Comment by loosebazooka** [comment]
fixed by #2402

**Comment by TadCordle** [comment]
@msoub @jontro We've released v2.2.0, which now respects the `DOCKER_CONFIG` environment variable.

**Comment by jontro** [comment]
Thanks! can confirm this now works out of the box. Great work
